compiler/src/iree/compiler/Codegen/Common/GPU/GPUPromoteMatmulOperands.cpp

```diff
@@ -24,11 +28,79 @@ namespace mlir::iree_compiler {
 #include "iree/compiler/Codegen/Common/GPU/Passes.h.inc"
 
 namespace {
+/// Helper to insert copy with derived thread config.
+Value promoteValue(OpBuilder &builder, Location loc, Value v) {
+  auto tensorType = cast<RankedTensorType>(v.getType());
+  SmallVector<OpFoldResult> mixedSizes = tensor::getMixedSizes(builder, loc, v);
+  Value empty = builder.create<tensor::EmptyOp>(loc, mixedSizes,
+                                                tensorType.getElementType());
+  auto copy = builder.create<linalg::CopyOp>(loc, v, empty);
+  setLoweringConfig(
+      copy, IREE::GPU::DerivedThreadConfigAttr::get(builder.getContext()));
+  return copy.getResult(0);
+}
+
+/// Helper to promote results.
```

Comment:
```suggestion
/// Helper to promote results. If the target value is consumed only by a `tensor.extract_slice`, this will promote the result of the slice instead.
```